File: N-th Tribonacci Number.go
func tribonacci(n int) int {
    if n < 2 {
        return n
    }

    if n == 2 {
        return 1
    }

    n1 := 0
    n2 := 1
    n3 := 1
    for i := 0; i < n-2; i++ {
        n4 := n1+n2+n3
        n1 = n2
        n2 = n3
        n3 = n4
    }
    return n3
}

File: N-th Tribonacci Number.py
class Solution:
    def tribonacci(self, n: int) -> int:
        if n <= 2:
            return [0, 1, 1][n]

        n1, n2, n3 = 0, 1, 1
        for _ in range(n - 2):
            n1, n2, n3 = n2, n3, n1 + n2 + n3
        return n3

